src/unity/extensions/additional_sframe_utilities.cpp

```diff
@@ -16,6 +16,55 @@
 using namespace turi;
 
 
+template <typename T>
+void copy_image_to_memory(const image_type& img, T *outptr,
+                          const std::vector<size_t>& outstrides,
+                          bool channel_last) {
+  ASSERT_EQ(outstrides.size(), 3);
+  size_t index_h, index_w, index_c;
+  if (channel_last) {
+    // Format: HWC
+    index_h = 0;
+    index_w = 1;
+    index_c = 2;
+  } else {
+    // Format: CHW
+    index_c = 0;
+    index_h = 1;
+    index_w = 2;
+  }
+
+  // Decode if needed
+  if (!img.is_decoded()) {
+    char* buf = NULL;
+    size_t length = 0;
+    if (img.m_format == Format::JPG) {
+      decode_jpeg((const char*)img.get_image_data(), img.m_image_data_size, &buf, length);
+    } else if (img.m_format == Format::PNG) {
```

Comment:
It seems like later you assume `buf` is not null. So this should be an `else` rather than a `else if`?

Perhaps the test here should just be an assert?

Reply:
Good point. The else should be unreachable (although that can easily change with the introduction of a new image type). I'll add an else-assert here.

Reply:
Done.